pkg/virt-controller/services/template.go

```diff
@@ -861,6 +864,10 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 		annotationsList[cniAnnotation] = cniNetworks
 	}
 
+	if HaveMasqueradeInterface(vmi.Spec.Domain.Devices.Interfaces) {
+		annotationsList[ISTIO_KUBEVIRT_ANNOTATION] = "k6t-eth0"
```

Comment:
This seems like a hardcoded name (eth0 part). I suspect at one point it may change, breaking this integration. We'll need functional tests to catch this future scenario.

Reply:
Functional will come in a separate PR because I need to create a provider for that.